### ISSUE TITLE
[bug] Inline links not sufficiently distinguishable from surrounding text in dark contexts

### ISSUE THREAD

**Issue by hellogreg** [RedHat-UX/red-hat-design-system#747]
### Describe the bug

By default, inline links in dark contexts are 1) not underlined (except when hovered/focused), and 2) do not have 3:1 color contrast from their surrounding paragraph text.

![image](https://user-images.githubusercontent.com/642695/221690919-9fb44da8-4e58-4cf8-99d2-b0055ff6dd3f.png)

This is [a failure of WCAG criterion 1.4.1: Use of color](https://www.w3.org/TR/WCAG20-TECHS/G183.html).

Frank Taylor first brought this up in the RHDS chat, [resulting from this Red Hat Jira issue](https://issues.redhat.com/browse/RHDX-3883).

### Which Component?

n/a

### Steps to reproduce

1. Go to a link using one of the dark mode contexts, like [the Red Hat Integration page](https://developers.redhat.com/products/integration/overview).
2. Scroll to a section of the page with the dark mode context until you see some text with inline links, like this:
![image](https://user-images.githubusercontent.com/642695/221689729-cf4d1ece-75db-49a5-a637-c02225360755.png)
3. Inspect the link text and the regular text and do a color contrast check ([WebAIM has an online checker](https://webaim.org/resources/contrastchecker/)). The regular text is #ffffff, and the link text is #73bcf7, for a contrast of 2.04:1. This is [below the required 3:1 contrast ratio](https://www.w3.org/TR/WCAG20-TECHS/G183.html).

### Expected behaviour

The link text should either 1) be underlined by default (pre-hover/focus), or 2) have 3:1 color contrast from the surrounding text (current fail) and also have a visual enhancement, like an underline, upon hover and focus (current pass). All normal-sized text must still meet or exceed 4.5:1 contrast from the background, too.

### Operating System (OS)

macOS

### Browser

Chrome

### More browser information

Chrome 110

### Additional Info

The tricky part of this may not be just a matter of picking a link color that contrasts with the non-link text. But it has to do so _while keeping its sufficient color contrast against all backgrounds_, too! The "dark" context could be especially tough.

Honestly, the easiest thing to do may just be to underline the inline links--at least in dark contexts.

**Comment by zeroedin** [comment]
This could be represented in docs if we aren't changing the color value with a best practices section under [foundations /typography/interaction states](https://ux.redhat.com/foundations/typography/)

Thoughts @coreyvickery?

Psuedo code example: (implementation would differ based on inside a component vs on general page)
```css
a {
  color: var(--_link-color, var(--rh-color-accent-base-on-light, #0066cc));
  text-decoration: var(--_link-decoration, none);
}

.dark a {
  --_link-color: var(--rh-color-accent-base-on-dark, #73bcf7);
  --_link-decoration: underline;
}
```

**Comment by bennypowers** [comment]
Do we need to fix this in tokens prior to 1.0? See also #869

**Comment by coreyvickery** [comment]
@bennypowers No, something so small doesn't need to hold up 1.0.

@zeroedin Makes sense to me, Marionne or I can update the documentation when we're ready to jump back into working on ux dot again.

@hellogreg We'll probably go with adding an underline for dark theme as Steven outlined.

**Comment by marionnegp** [comment]
This should be resolved once we add the dashed underline style for inline links. Adding a underline to inline links in our Figma library is in review.

Here’s where links are being underlined in our library:

- Card - asset
- Card - list
- Card - video
- Footnote
- Form: kept “privacy statement” underlined for all variants for consistency and to distinguish it from link buttons nearby.
- List
- Tag - linked

There will also be text styles for links in light and dark theme.

_A note for the dotted underline style in Figma:_ I used a 1.5px underline thickness because 1px looks way too thin with dotted lines. It was kind of hard for me to look at on my screen. :woman-shrugging::skin-tone-4: This shouldn’t affect use of 1px dashed underlines that will be used in code.

**Comment by marionnegp** [comment]
This was closed by #2418
